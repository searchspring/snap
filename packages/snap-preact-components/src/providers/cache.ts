export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
