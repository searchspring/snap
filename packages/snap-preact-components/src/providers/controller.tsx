import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
