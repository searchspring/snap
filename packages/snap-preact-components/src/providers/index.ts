export * from './store';
export * from './theme';
