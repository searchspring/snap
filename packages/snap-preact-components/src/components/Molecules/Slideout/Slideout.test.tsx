import { h } from 'preact';

import { render } from '@testing-library/preact';
import userEvent from '@testing-library/user-event';

import { Slideout, SlideDirectionType } from './Slideout';
import { ThemeProvider } from '../../../providers';

describe('Slideout Component', () => {
	beforeEach(() => {
		Object.defineProperty(window, 'matchMedia', {
			writable: true,
			value: jest.fn().mockImplementation((query) => ({
				matches: true,
				media: query,
				onchange: null,
				addListener: jest.fn(), // Deprecated
				removeListener: jest.fn(), // Deprecated
				addEventListener: jest.fn(),
				removeEventListener: jest.fn(),
				dispatchEvent: jest.fn(),
			})),
		});
	});

	const theme = {
		components: {
			slideout: {
				width: '550px',
			},
		},
	};

	it('renders and is visible', () => {
		const args = {
			active: true,
		};
		const rendered = render(<Slideout {...args} />);
		const slideoutElement = rendered.container.querySelector('.ss__slideout')!;
		const styles = getComputedStyle(slideoutElement);
		expect(slideoutElement).toBeInTheDocument();
		expect(styles.left).toBe('0px');
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is not visible', () => {
		const args = {
			active: false,
		};
		const rendered = render(<Slideout {...args} />);
		const slideoutElement = rendered.container.querySelector('.ss__slideout')!;
		const styles = getComputedStyle(slideoutElement);
		expect(styles.left).toBe('-300px');
		expect(styles.left).not.toBe('0px');
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('has custom width', () => {
		const args = {
			active: false,
			width: '400px',
		};
		const rendered = render(<Slideout {...args} />);
		const slideoutElement = rendered.container.querySelector('.ss__slideout')!;
		const styles = getComputedStyle(slideoutElement);
		expect(styles.maxWidth).toBe(args.width);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('has custom displayAtWidth', () => {
		const args = {
			active: false,
			width: '400px',
			displayAt: '(min-width: 600px)',
		};
		const rendered = render(<Slideout {...args} />);
		const slideoutElement = rendered.container.querySelector('.ss__slideout');
		expect(slideoutElement).toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('does not render if matchMedia returns false', () => {
		Object.defineProperty(window, 'matchMedia', {
			writable: true,
			value: jest.fn().mockImplementation((query) => ({
				matches: false, // return false
				media: query,
				onchange: null,
				addListener: jest.fn(), // Deprecated
				removeListener: jest.fn(), // Deprecated
				addEventListener: jest.fn(),
				removeEventListener: jest.fn(),
				dispatchEvent: jest.fn(),
			})),
		});
		const args = {
			active: false,
		};
		const rendered = render(<Slideout {...args} />);
		const slideoutElement = rendered.container.querySelector('.ss__slideout');
		expect(slideoutElement).not.toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('displays overlay', () => {
		const args = {
			active: true,
		};
		const rendered = render(<Slideout {...args} />);
		const overlayElement = rendered.container.querySelector('.ss__overlay.ss__overlay--active');
		expect(overlayElement).toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('hides overlay when inactive', () => {
		const args = {
			active: false,
		};
		const rendered = render(<Slideout {...args} />);
		const overlayElement = rendered.container.querySelector('.ss__overlay.ss__overlay--active');
		expect(overlayElement).not.toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('has custom overlay color', () => {
		const args = {
			active: true,
			overlayColor: 'rgba(0, 0, 0, 0.1)',
		};
		const rendered = render(<Slideout {...args} />);
		const overlayElement = rendered.container.querySelector('.ss__overlay.ss__overlay--active')!;
		const styles = getComputedStyle(overlayElement);
		expect(styles.background).toBe(args.overlayColor);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can set custom slideDirection', () => {
		const args = {
			active: true,
			slideDirection: 'right' as SlideDirectionType,
		};
		const rendered = render(<Slideout {...args} />);
		const overlayElement = rendered.container.querySelector('.ss__slideout.ss__slideout--active')!;
		const styles = getComputedStyle(overlayElement);
		expect(styles.right).toBe('0px');
		expect(styles.left).toBe('');
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can set custom transitionSpeed', () => {
		const args = {
			active: true,
			transitionSpeed: '0.55s',
		};
		const rendered = render(<Slideout {...args} />);
		const overlayElement = rendered.container.querySelector('.ss__slideout.ss__slideout--active')!;
		const styles = getComputedStyle(overlayElement);
		expect(styles.transition).toBe(`left ${args.transitionSpeed}`);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('renders a wrapper element with classname', () => {
		const args = {
			active: true,
			buttonContent: 'click me',
		};
		const rendered = render(<Slideout {...args} />);

		const buttonElement = rendered.container.querySelector('.ss__slideout__button');
		expect(buttonElement).toBeInTheDocument();
		expect(buttonElement).toHaveTextContent(args.buttonContent);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('does not render a wrapper element when using "noButtonWrapper" prop', () => {
		const args = {
			noButtonWrapper: true,
			className: 'clickit',
			buttonContent: <div>click me</div>,
		};
		const rendered = render(<Slideout {...args} />);

		const buttonWrapperElement = rendered.container.querySelector('.ss__slideout__button');
		const buttonElement = rendered.container.querySelector(`.${args.className}`);
		expect(buttonWrapperElement).not.toBeInTheDocument();
		expect(buttonElement).toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('toggles the "ss__slideout--active" class when the button is clicked', async () => {
		const args = {
			buttonContent: 'click me',
		};
		const rendered = render(<Slideout {...args} />);

		const buttonElement = rendered.container.querySelector('.ss__slideout__button');
		const containerElement = rendered.container.querySelector('.ss__slideout');

		expect(containerElement).not.toHaveClass('ss__slideout--active');
		expect(rendered.asFragment()).toMatchSnapshot();

		// click the button
		if (buttonElement) await userEvent.click(buttonElement);
		expect(containerElement).toHaveClass('ss__slideout--active');
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('toggles the "ss__slideout--active" class when using clonedElement props', async () => {
		const ButtonComponent = (props: any) => {
			return (
				<div className="custom-button" onClick={() => props.toggleActive()}>
					{props.active ? 'active' : 'inactive'}
				</div>
			);
		};

		const args = {
			noButtonWrapper: true,
			buttonContent: <ButtonComponent />,
		};

		const rendered = render(<Slideout {...args} />);
		const buttonElement = rendered.container.querySelector('.custom-button');
		const containerElement = rendered.container.querySelector('.ss__slideout');

		expect(containerElement).not.toHaveClass('ss__slideout--active');
		expect(buttonElement).toHaveTextContent('inactive');
		expect(rendered.asFragment()).toMatchSnapshot();

		// click the button
		if (buttonElement) await userEvent.click(buttonElement);
		expect(containerElement).toHaveClass('ss__slideout--active');
		expect(buttonElement).toHaveTextContent('active');
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can disable styles', () => {
		const args = {
			active: true,
			disableStyles: true,
		};
		const rendered = render(<Slideout {...args} />);

		const resultElement = rendered.container.querySelector('.ss__slideout');

		expect(resultElement?.classList).toHaveLength(2);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can add additional styles', () => {
		const args = {
			disableStyles: true,
			style: {
				backgroundColor: 'green',
			},
		};

		const rendered = render(<Slideout {...args} />);
		const slideoutElement = rendered.container.querySelector('.ss__slideout');
		const styles = getComputedStyle(slideoutElement!);

		expect(styles.backgroundColor).toBe(args.style.backgroundColor);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable with ThemeProvider', () => {
		const args = {
			active: true,
		};
		const rendered = render(
			<ThemeProvider theme={theme}>
				<Slideout {...args} />
			</ThemeProvider>
		);
		const slideoutElement = rendered.container.querySelector('.ss__slideout')!;
		const styles = getComputedStyle(slideoutElement);
		expect(styles.maxWidth).toBe(theme.components.slideout.width);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable with theme prop', () => {
		const args = {
			active: true,
		};
		const rendered = render(<Slideout {...args} theme={theme} />);
		const slideoutElement = rendered.container.querySelector('.ss__slideout')!;
		const styles = getComputedStyle(slideoutElement);
		expect(styles.maxWidth).toBe(theme.components.slideout.width);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable with theme  prop overrides ThemeProvider', () => {
		const args = {
			active: true,
		};
		const themeOverride = {
			components: {
				slideout: {
					width: '600px',
				},
			},
		};
		const rendered = render(
			<ThemeProvider theme={theme}>
				<Slideout {...args} theme={themeOverride} />
			</ThemeProvider>
		);
		const slideoutElement = rendered.container.querySelector('.ss__slideout')!;
		const styles = getComputedStyle(slideoutElement);
		expect(styles.maxWidth).toBe(themeOverride.components.slideout.width);
		expect(rendered.asFragment()).toMatchSnapshot();
	});
});
