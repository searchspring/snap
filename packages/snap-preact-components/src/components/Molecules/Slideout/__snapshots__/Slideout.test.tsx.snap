// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideout Component can add additional styles 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  background-color: green;
}

<div
    class="ss__slideout emotion-0"
  />
  <div
    class="ss__overlay ss__slideout__overlay"
  />
</DocumentFragment>
`;

exports[`Slideout Component can disable styles 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  <div
    class="ss__slideout ss__slideout--active"
  />
  <div
    class="ss__overlay ss__overlay--active ss__slideout__overlay"
  />
</DocumentFragment>
`;

exports[`Slideout Component can set custom slideDirection 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: right 0.25s;
  transition: right 0.25s;
  left: initial;
  right: 0;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component can set custom transitionSpeed 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.55s;
  transition: left 0.55s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.55s ease 0s,left 0s ease 0.55s;
  transition: background 0.55s ease 0s,left 0s ease 0.55s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.55s ease,left 0s ease;
  transition: background 0.55s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component displays overlay 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component does not render a wrapper element when using "noButtonWrapper" prop 1`] = `
<DocumentFragment>
  <div>
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: -300px;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout clickit emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component does not render if matchMedia returns false 1`] = `<DocumentFragment />`;

exports[`Slideout Component has custom displayAtWidth 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: -400px;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 400px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component has custom overlay color 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0, 0, 0, 0.1);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component has custom width 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: -400px;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 400px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component hides overlay when inactive 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: -300px;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component is not visible 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: -300px;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 550px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component is themeable with theme  prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 600px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component is themeable with theme prop 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 550px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component renders a wrapper element with classname 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component renders and is visible 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component toggles the "ss__slideout--active" class when the button is clicked 1`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: -300px;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component toggles the "ss__slideout--active" class when the button is clicked 2`] = `
<DocumentFragment>
  <div
    class="ss__slideout__button"
  >
    click me
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component toggles the "ss__slideout--active" class when using clonedElement props 1`] = `
<DocumentFragment>
  <div
    class="custom-button"
  >
    inactive
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: -300px;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;

exports[`Slideout Component toggles the "ss__slideout--active" class when using clonedElement props 2`] = `
<DocumentFragment>
  <div
    class="custom-button"
  >
    active
  </div>
  .emotion-0 {
  display: block;
  position: fixed;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  left: 0;
  right: initial;
  bottom: initial;
  top: 0;
  height: 100%;
  z-index: 10004;
  width: 90%;
  max-width: 300px;
  padding: 10px;
  background: #fff;
  box-sizing: border-box;
  overflow-y: auto;
}

<div
    class="ss__slideout ss__slideout--active emotion-0"
  />
  .emotion-0 {
  -webkit-transition: background 0.25s ease 0s,left 0s ease 0.25s;
  transition: background 0.25s ease 0s,left 0s ease 0.25s;
  position: fixed;
  z-index: 10003;
  height: 100%;
  width: 100%;
  top: 0;
  left: -100%;
}

.emotion-0.ss__overlay--active {
  -webkit-transition: background 0.25s ease,left 0s ease;
  transition: background 0.25s ease,left 0s ease;
  background: rgba(0,0,0,0.8);
  left: 0;
}

<div
    class="ss__overlay ss__overlay--active ss__slideout__overlay emotion-0"
  />
</DocumentFragment>
`;
